docs/DeveloperGuide.md

```diff
@@ -521,330 +523,322 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-3.  User requests to delete a specific employee in the list
-4.  SudoHR deletes the employee
+1.  User requests to add a new employee to SudoHR.
+2.  SudoHR adds the employee.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. There is already an employee in SudoHR with the same eid, phone number or email as the new employee added.
 
-  Use case ends.
+    * 1a1. SudoHR shows an error message.
 
-* 3a. The given index is invalid.
+      Use case resumes at step 1.
 
-    * 3a1. SudoHR shows an error message.
+* 1b. The given command has invalid inputs.
 
-      Use case resumes at step 2.
+    * 1b1. SudoHR shows an error message.
 
-**Use case: UC1 - Create a department**
+     Use case resumes at step 1.
 
-**MSS:**
-1. User requests to create a department.
-2. SudoHR creates the department.
+* 1c. The given command has missing inputs.
 
-    Use case ends.
+    * 1c1. SudoHR shows an error message.
 
-**Extensions**
+      Use case resumes at step 1.
 
-* 1a. The given argument is invalid.
 
-    * 1a1. SudoHR shows an error message.
 
-      Use case resumes at step 1.
+**Use case: UC2 - Delete a employee**
 
-**Use case: UC2 - Add employee to department**
-**MSS:**
-1. User lists department
-2. User requests to add employee to the department.
-3. SudoHR adds employee to the department.
-4. Use case ends.
+**MSS**
 
-**Extensions**
+1.  User requests SudoHR to __list all employees(UC5)__.
+2.  User requests to delete a specific employee with a specific eid.
+3.  SudoHR deletes the employee.
 
-* 1a. The given argument is invalid.
+    Use case ends.
 
-    * 1a1. SudoHR shows an error message.
+**Extensions**
 
-      Use case resumes at step 2.
+* 1a. The list is empty.
 
-**Use case: UC3 - List Employees in department**
+      Use case ends.
 
-**MSS:**
-1. User requests to list the employees in a specified department.
-2. SudoHr lists all the employees in the department.
+* 2a. There is no employee with the eid given as input.
 
-   Use case ends.
+    * 2a1. SudoHR shows an error message.
 
-**Extensions**
+      Use case resumes at step 2
 
-* 1a. The given department is invalid.
+* 2b. The given command has invalid inputs.
 
-    * 1a1. SudoHR shows an error message.
+    * 2b1. SudoHR shows an error message.
 
-      Use case resumes at step 1.
+      Use case resumes at step 2.
 
-**Use case: UC4 - List Employees in a department present on a given day.**
+* 2c. The given command has missing inputs.
 
-**MSS:**
-1. User requests to list all employees in a given department who are present.
-2. SudoHR lists the employees.
+    * 2c1. SudoHR shows an error message.
 
-   Use case ends.
+      Use case resumes at step 2.
 
-**Extensions**
 
-* 1a. The given department does not exist.
+**Use case: UC3 - Edit a employee**
 
-    * 1a1. SudoHR shows an error message.
+**MSS**
 
-      Use case resumes at step 1.
+1.  User requests SudoHR to __show the details of a employee with a specific eid(UC4)__.
```

Comment:
I think we dont have to show the employee details for us to be able to edit the employee, as long as we know what the eid is. Same concern as the comment about us changing the implementation to use fixed, eid instead of dynamic indexes.